### ISSUE TITLE
When using ConvertImageDtype to convert images in the CUB dataset to torch.uint8, strange results are obtained.

### ISSUE THREAD

**Issue by deemoe404** [sicara/easy-few-shot-learning#137]
**Problem**
I attempted to apply data augmentation (`RandAugment` from `torchvision.transforms`) to the data in the CUB dataset. However, this method requires the tensor format to be `uint8`, whereas the images obtained through the `DataLoader` are in `float32` format.

**Considered solutions**
I tried using `ConvertImageDtype` from `torchvision.transforms` to change the format, but ended up with strange results.

```
img = ConvertImageDtype(torch.uint8)(example_support_images[0])

augmenter = RandAugment()
imgs = [augmenter(img) for _ in range(4)]

plot_images(imgs, "augmented support images", images_per_row=4)
```

![output](https://github.com/sicara/easy-few-shot-learning/assets/90408077/d0c57556-5366-4bea-98d5-3978d6a68fe2)

I think I might have misunderstood something, but I'm not sure where I went wrong.

**How can we help**
Am I on the right track with my approach to data augmentation? I'm not very familiar with this field, so any guidance would be greatly appreciated! 😭

**Comment by ebennequin** [comment]
I see no indication in the documentation that `RandAugment` wouldn't work with `float32` tensors. I have successfully used it with images directly taken out of the CUB dataset:

```python
from matplotlib import pyplot as plt
from torchvision.transforms.v2 import RandAugment

from build.lib.easyfsl.utils import plot_images
from easyfsl.datasets import CUB

dataset = CUB(split="test", training=False)
original_image = dataset[0][0]

plot_images(original_image.unsqueeze(0), "original normalized image", 1)
plt.show()

augmenter = RandAugment()
augmented_images = [augmenter(original_image) for _ in range(4)]

plot_images(augmented_images, "augmented images", images_per_row=4)
plt.show()
```

This gives:
![image](https://github.com/sicara/easy-few-shot-learning/assets/45969336/103eae1d-3ccf-4160-84c7-a36e7d9375c7)
![image](https://github.com/sicara/easy-few-shot-learning/assets/45969336/459740b2-3ade-4c0a-8ed2-7efad0d8cce7)

Did you import `RandAugment` from `torchvision.transforms.v2`?

**Comment by deemoe404** [comment]
I read the documentation [here](https://pytorch.org/vision/main/generated/torchvision.transforms.v2.RandAugment.html#torchvision.transforms.v2.RandAugment), and it says `If the input is torch.Tensor, it should be of type torch.uint8,` 😭

Anyway, using the images without conversion directly did works fine. I apologize for the inconvenience caused by this minor issue, and I appreciate your assistance greatly!

**Comment by deemoe404** [comment]
> I see no indication in the documentation that `RandAugment` wouldn't work with `float32` tensors. I have successfully used it with images directly taken out of the CUB dataset:
> 
> ```python
> from matplotlib import pyplot as plt
> from torchvision.transforms.v2 import RandAugment
> 
> from build.lib.easyfsl.utils import plot_images
> from easyfsl.datasets import CUB
> 
> dataset = CUB(split="test", training=False)
> original_image = dataset[0][0]
> 
> plot_images(original_image.unsqueeze(0), "original normalized image", 1)
> plt.show()
> 
> augmenter = RandAugment()
> augmented_images = [augmenter(original_image) for _ in range(4)]
> 
> plot_images(augmented_images, "augmented images", images_per_row=4)
> plt.show()
> ```
> 
> This gives: ![image](https://private-user-images.githubusercontent.com/45969336/309236179-377bf70e-1b3d-43cf-a15d-946c50697a25.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDkyOTIxNjMsIm5iZiI6MTcwOTI5MTg2MywicGF0aCI6Ii80NTk2OTMzNi8zMDkyMzYxNzktMzc3YmY3MGUtMWIzZC00M2NmLWExNWQtOTQ2YzUwNjk3YTI1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzAxVDExMTc0M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg4MDMyZDI3ZDhkZWVmZDdiMTg1MzFiYmYzN2M1ZjQxMThlNzUzZGUyZTJkY2NhY2M5ODNhNDk1ODQ0ZDljMTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.AbxTQFVd4Y9dYpnS9Xm57wPKcimKLwH4lK6pVpk2W18) ![image](https://private-user-images.githubusercontent.com/45969336/309236218-265382d3-c13d-45f7-9dfb-6150bc60cdc3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDkyOTIxNjMsIm5iZiI6MTcwOTI5MTg2MywicGF0aCI6Ii80NTk2OTMzNi8zMDkyMzYyMTgtMjY1MzgyZDMtYzEzZC00NWY3LTlkZmItNjE1MGJjNjBjZGMzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzAxVDExMTc0M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE0NzBmZDYxZTNkOWViZGZmMjA5MGU4NWY0Yjk3ZjQ3YzJjYWFmMWNjMTFmM2RhNzFhMjlhNzE5ZWFjYWY4ZDMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.rEOFfTZ4Zgr7CnzbIvjQpecFuDKdRTpM29JsO-XVNxw)
> 
> Did you import `RandAugment` from `torchvision.transforms.v2`?

By the way, regarding the second image, are the results for positions 1, 3, and 4 normal during data augmentation? While experimenting with this augmentation function, I occasionally observe distorted images in the results, and sometimes they appear normal.

**Comment by ebennequin** [comment]
I am not familiar with the transformations used by `RandAugment` so can't help you more than the documentation.

Note that the original image delivered by the `CUB` dataset is normalized on ImageNet mean and variance, so it already looks "funny".
